### DISCUSSION TITLE
Setting Heading of IconSymbol or PlaceNode

### DISCUSSION THREAD

**Discussion by nebsar** [pelicanmapping/osgearth#1966] [Support Q&A] [answered]
Is there any way to set heading of IconSymbol or PlaceNode other than 0? I do not want it to stay always upright. I want to rotate it.  Thanks.

Edit: I found that IconSymbol heading can be set before creating the PlaceNode. But after creating the PlaceNode, I cannot change the heading of the IconSymbol. How can I change the heading dynamically?

**Comment by gwaldron** [comment]
No, the API has no support for that; sorry.

**Reply by nebsar** [reply]
Thanks Glenn. As I see from the old forum and here, there is a lot of request for this feature. My 2 cents is, it would be a good feature to update the heading of the PleaceNode on the fly.
Thanks.

**Comment by keince** [comment, marked as answer]
Have you tried the following in an update callback for your PlaceNode?

```
auto style = place->getStyle();
style.getOrCreate<IconSymbol>()->heading() = headingDeg;
place->setStyle(style);
```

This works for us in rotating the icon of a PlaceNode based on a heading. Just be aware that calling "setStyle(...)" forces a recompile of the object using it so it's expensive. Only call it when the heading changes.

-S

**Reply by nebsar** [reply]
Ok thank you very much.
